lib/json-schema/util/uri.rb

```diff
@@ -87,8 +87,11 @@ def self.strip_fragment(uri)
 
       def self.file_uri(uri)
         parsed_uri = parse(uri)
-
-        Addressable::URI.convert_path(parsed_uri.path)
+        if not parsed_uri.scheme.nil?
+          parsed_uri
+        else
```

Comment:
The conditional would read better as `if parsed_uri.scheme`

Reply:
Yep, makes sense. I'll edit the commit.